plutus-tx/src/PlutusTx/AssocMap.hs

```diff
@@ -95,24 +97,26 @@ instance (Haskell.Ord k, Haskell.Ord v) => Haskell.Ord (Map k v) where
   Map l <= Map r =
     on (Haskell.<=) HMap.fromList l r
 
--- | Hand-written instances to use the underlying 'Map' type in 'Data', and
--- to be reasonably efficient.
+{-| Hand-written instances to use the underlying 'Map' type in 'Data', and
+to be reasonably efficient.
+-}
 instance (ToData k, ToData v) => ToData (Map k v) where
   toBuiltinData (Map es) = BI.mkMap (mapToBuiltin es)
-    where
-      {-# INLINE mapToBuiltin #-}
-      mapToBuiltin :: [(k, v)] -> BuiltinList (BuiltinPair BI.BuiltinData BI.BuiltinData)
-      mapToBuiltin = go
-        where
-          go :: [(k, v)] -> BuiltinList (BuiltinPair BI.BuiltinData BI.BuiltinData)
-          go []            = P.mkNil
-          go ((k, v) : xs) = BI.mkCons (BI.mkPairData (toBuiltinData k) (toBuiltinData v)) (go xs)
-
--- | A hand-written transformation from 'Data' to 'Map'. Compared to 'unsafeFromBuiltinData',
--- it is safe to call when it is unknown if the 'Data' is built with 'Data's 'Map' constructor.
--- Note that it is, however, unsafe in the sense that it assumes that any map
--- encoded in the 'Data' is well-formed, i.e. 'fromBuiltinData' does not perform any
--- deduplication of keys or of key-value pairs!
+   where
+    {-# INLINE mapToBuiltin #-}
+    mapToBuiltin :: [(k, v)] -> BuiltinList (BuiltinPair BI.BuiltinData BI.BuiltinData)
+    mapToBuiltin = go
+     where
+      go :: [(k, v)] -> BuiltinList (BuiltinPair BI.BuiltinData BI.BuiltinData)
+      go []            = P.mkNil
+      go ((k, v) : xs) = BI.mkCons (BI.mkPairData (toBuiltinData k) (toBuiltinData v)) (go xs)
+
+{-| A hand-written transformation from 'Data' to 'Map'. Compared to 'unsafeFromBuiltinData',
+it is safe to call when it is unknown if the 'Data' is built with 'Data's 'Map' constructor.
+Note that it is, however, unsafe in the sense that it assumes that any map
+encoded in the 'Data' is well-formed, i.e. 'fromBuiltinData' does not perform any
+deduplication of keys or of key-value pairs!
+-}
```

Comment:
This one used to relate to the `mapToBuiltin`. It should probably be a regular comment, not a haddock.

Reply:
https://github.com/IntersectMBO/plutus/blob/541aff5c501179cce9c419bc27bc49c0a8ef9a81/plutus-tx/src/PlutusTx/AssocMap.hs#L111

This looks correct. Was this incorrectly placed originally?

Reply:
Nevermind, I got confused.  It does relate to the instance below.